### ISSUE TITLE
Invalid check for Redirect URI: "Must be Absolute"

### ISSUE THREAD

**Issue by bshaffer** [google/google-auth-library-php#54]
The OAuth2 library is requiring that [the redirect URI must be absolute](https://github.com/google/google-auth-library-php/blob/master/src/OAuth2.php#L626), which sounds good except in the case of [installed applications](https://developers.google.com/identity/protocols/OAuth2InstalledApp), where the redirect URI can be `urn:ietf:wg:oauth:2.0:oob` or `urn:ietf:wg:oauth:2.0:oob:auto`

We could whilelist the `urn` prefix, or whitelist these two strings only. Either way, this will need to be fixed to be consistent with the [Google Auth documentation](https://developers.google.com/identity/protocols/OAuth2InstalledApp#choosingredirecturi)

**Comment by bshaffer** [comment]
The [spec](https://tools.ietf.org/html/rfc6749#section-3.1.2) states 

    The redirection endpoint URI MUST be an absolute URI as defined by RFC3986 Section 4.3.`

[section-4.3](https://tools.ietf.org/html/rfc3986#section-4.3) states

    absolute-URI  = scheme ":" hier-part [ "?" query ]

And [section-3](https://tools.ietf.org/html/rfc3986#section-3) defines `hier-part` even further

All of this means the URI could match any of the valid [urn namespaces](http://www.w3.org/TR/uri-clarification/#urn-namespaces) and still be considered absolute. Unfortunately, the call to Guzzle's [`Url::isAbsolute`](https://github.com/guzzle/guzzle/blob/5.3/src/Url.php#L494) requires `sceme+host`, while the call to PHP's `parse_url` gives us a `sceme+path`.

So, long story short, we should just check for `scheme+path` after checking `isAbsolute`, and consider switching to the newer [`guzzle/psr7`](https://github.com/guzzle/psr7/blob/master/src/Uri.php) library at some point.
